### PR TITLE
Add semver parsing to database check for PostgreSQL and PostGIS

### LINKED ISSUE #1: Postgres version check fails < 10.0
### Context
At boot, the version checker fails because it doesn't account for patch versions. 

`cargo run` fails with:

```
ERROR: Connection unable obtain postgres version using (READ/WRITE) hecate@localhost:5432/hecate
ERROR: database error
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:378:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

The SQL select statement for version check fails with:
```
ERROR:  invalid input syntax for type double precision: "9.6.14"
```

cc/ @ingalls

### PR DESCRIPTION
**Pull request by guidorice**
### Context
Fixes #1

The `version()` check for PostgreSQL and PostGIS fails on 9.x because the value cannot always be parsed as a float, e.g. `9.6.16`.  Here is an example error from `psql`

```sql
SELECT
 (regexp_matches(version(), 'PostgreSQL (.*?) '))[1]::FLOAT AS postgres_v,
 (regexp_matches(postgis_version(), '^(.*?) '))[1]::FLOAT AS postgis_v;
-- ERROR:  invalid input syntax for type double precision: "9.6.16"
```

This PR adds semantic versioning parsing and also supports the edge case where PostgreSQL or PostGIS return a partial semver, e.g. `12.1`.

If you run this against PostgreSQL 9, you should get

> thread 'main' panicked at 'ERROR: Hecate requires a min PostgreSQL version of >= 11.0.0, got 9.6.16.', hecate/src/main.rs:157:29

I also bumped the required PostgreSQL and PostGIS versions to what seems best.

cc/ @ingalls

### CODE REVIEW

File: hecate/src/main.rs [hunk 1]
```diff
@@ -134,29 +136,49 @@ fn database_check(conn_str: &String, is_read: bool) {
...
+                            println!("ERROR: Connection unable obtain PostgreSQL version using ({}) {}", conn_type, conn_str);
                             std::process::exit(1);
                         }
-
-                        let postgres_v: f64 = res.get(0).get(0);
-                        if postgres_v < hecate::POSTGRES {
-                            println!("ERROR: Hecate requires a min postgres version of {}", hecate::POSTGRES);
-                            std::process::exit(1);
+                        let postgres_v: String = res.get(0).get(0);
+                        let want_postgres_v = VersionReq::parse(hecate::POSTGRES_VERSION).unwrap();
```
Comment (ingalls) [inline hecate/src/main.rs]: Let's add a `match` statement here with a human readable error message if it fails. (Obviously it shouldn't - but I fat finger things all the time)

File: hecate/src/main.rs [hunk 2]
```diff
@@ -134,29 +136,49 @@ fn database_check(conn_str: &String, is_read: bool) {
...
+                        let got_postgres_v = match Version::parse(&postgres_v){
+                            Ok(version) => version,
+                            Err(_) => {
+                                let fix_semver = format!("{}.0", postgres_v);
+                                Version::parse(&fix_semver).unwrap()
```
Comment (ingalls) [inline hecate/src/main.rs]: See above, just on a bit of a warpath to remove as many `unwraps` as I can unless they are clearly 100% safe. Nick a year ago used it far to often.

**Comment by guidorice** [comment]
@ingalls good on you about the `unwrap` calls! I switched them to `expect` which is basically the same but with an error message. But not as much code as a custom match statement.

I also made some other improvements to the Semver parsing. Let me know what you think!

File: hecate/src/lib.rs [hunk 3]
```diff
@@ -345,7 +354,7 @@ fn server(
     auth::check(&auth_rules.0.server, auth::RW::Read, &auth)?;
 
     Ok(Json(json!({
-        "version": VERSION,
+        "version": env!("CARGO_PKG_VERSION"),
```
Comment (ingalls) [inline hecate/src/lib.rs]: Well this is awesome.

**Pull request closed without merging**